addons/mail/tests/common.py

```diff
@@ -135,9 +135,8 @@ def format(self, template, to='groups@example.com, other@gmail.com', subject='Fr
 
     def format_and_process(self, template, email_from, to, subject='Frogs', cc='',
                            return_path='', extra='',  msg_id=False,
-                           model=None, target_model='mail.test.gateway', target_field='name',
+                           model=None, target_model=None,
                            **kwargs):
-        self.assertFalse(self.env[target_model].search([(target_field, '=', subject)]))
```

Comment:
Actually it was also to be a bit less dependent on message_process implementation. However as it did not change in a lot of years, maybe this check is useless, not sure. But globally it ensures we create a new record instead of finding an existing one "by mistake / due to test writing". Not sure of the impact if we change it (especially those searches should be quite straighforward as we mainly use test models with few records in db).

Or we do both, check record and use message_process returned value.